R/pkg/R/DataFrame.R

```diff
@@ -1053,6 +1057,13 @@ setMethod("collect",
                   df[[colIndex]] <- col
                 } else {
                   colType <- dtypes[[colIndex]][[2]]
+                  if (is.null(PRIMITIVE_TYPES[[colType]])) {
+                    specialtype <- specialtypeshandle(colType)
+                    if (!is.null(specialtype)) {
+                      colType <- specialtype
```

Comment:
in this case colType is supposed to be JVM type, so we are mixing in what we are using the function for?